Panic on nil *request.Url

### retryablehttp version:
latest

### Current Behavior:
Panic on nil pointer *URL

### Expected Behavior:
Proper checks in place

### Steps To Reproduce:
katana integration tests:
```
$ ./run.sh 
::group::Build katana
::endgroup::
::group::Build katana integration-test
::endgroup::
Running test cases for "code"
2023/03/23 09:43:23 <nil>
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x14f3ddc]

goroutine 46 [running]:
github.com/projectdiscovery/utils/url.(*URL).Update(...)
        /Users/marcornvh/go/src/github.com/projectdiscovery/utils/url/url.go:55
github.com/projectdiscovery/retryablehttp-go.(*Request).Update(0xc0018a1bc0?)
        /Users/marcornvh/go/pkg/mod/github.com/projectdiscovery/retryablehttp-go@v1.0.13/request.go:106 +0x1c
github.com/projectdiscovery/retryablehttp-go.(*Request).Clone(0xc001798f60, {0x1d081d0, 0xc00018c010})
        /Users/marcornvh/go/pkg/mod/github.com/projectdiscovery/retryablehttp-go@v1.0.13/request.go:119 +0x2c
github.com/projectdiscovery/retryablehttp-go.(*Request).Dump(0xc0018a1da8?)
        /Users/marcornvh/go/pkg/mod/github.com/projectdiscovery/retryablehttp-go@v1.0.13/request.go:144 +0x2b
github.com/projectdiscovery/katana/pkg/engine/standard.(*Crawler).makeRequest(0xc00051c7c0, 0xc001677400, 0xc0017a48c0)
        /Users/marcornvh/go/src/github.com/projectdiscovery/katana/pkg/engine/standard/crawl.go:60 +0x768
github.com/projectdiscovery/katana/pkg/engine/common.(*Shared).Do.func1()
        /Users/marcornvh/go/src/github.com/projectdiscovery/katana/pkg/engine/common/base.go:207 +0xeb
created by github.com/projectdiscovery/katana/pkg/engine/common.(*Shared).Do
        /Users/marcornvh/go/src/github.com/projectdiscovery/katana/pkg/engine/common/base.go:197 +0x3ab
```

--- a/request.go
+++ b/request.go
@@ -171,6 +171,14 @@ func FromRequest(r *http.Request) (*Request, error) {
 		Auth:    nil,
 	}
 
+	if r.URL != nil {
+		urlx, err := urlutil.Parse(r.URL.String())
+		if err != nil {
+			return nil, err
+		}
+		req.URL = urlx
+	}
+
 	if r.Body != nil {
 		body, err := readerutil.NewReusableReadCloser(r.Body)
 		if err != nil {

